pkg/scheduler/plugins/capacity/capacity.go

```diff
@@ -0,0 +1,332 @@
+		// Considering a Spark job is completed(driver pod is completed) while the podgroup keeps running, the allocated resource will be reserved again if without the judgement.
+		if job.PodGroup.Status.Phase == scheduling.PodGroupRunning &&
+			job.PodGroup.Spec.MinResources != nil &&
+			int32(util.CalculateAllocatedTaskNum(job)) >= job.PodGroup.Spec.MinMember {
+			inqueued := util.GetInqueueResource(job, job.Allocated)
+			attr.inqueue.Add(inqueued)
+		}
+		attr.elastic.Add(job.GetElasticResources())
+		klog.V(5).Infof("Queue %s allocated <%s> request <%s> inqueue <%s> elastic <%s>",
+			attr.name, attr.allocated.String(), attr.request.String(), attr.inqueue.String(), attr.elastic.String())
+	}
+
+	for _, attr := range cp.queueOpts {
+		if attr.realCapability != nil {
+			attr.deserved.MinDimensionResource(attr.realCapability, api.Infinity)
+		}
+		attr.deserved.MinDimensionResource(attr.request, api.Zero)
+
+		attr.deserved = helpers.Max(attr.deserved, attr.guarantee)
+		cp.updateShare(attr)
+		klog.V(4).Infof("The attributes of queue <%s> in capacity: deserved <%v>, realCapability <%v>, allocate <%v>, request <%v>, elastic <%v>, share <%0.2f>",
+			attr.name, attr.deserved, attr.realCapability, attr.allocated, attr.request, attr.elastic, attr.share)
+	}
+
+	ssn.AddQueueOrderFn(cp.Name(), func(l, r interface{}) int {
+		lv := l.(*api.QueueInfo)
+		rv := r.(*api.QueueInfo)
+
+		if cp.queueOpts[lv.UID].share == cp.queueOpts[rv.UID].share {
+			return 0
+		}
+
+		if cp.queueOpts[lv.UID].share < cp.queueOpts[rv.UID].share {
+			return -1
+		}
+
+		return 1
+	})
+
+	ssn.AddReclaimableFn(cp.Name(), func(reclaimer *api.TaskInfo, reclaimees []*api.TaskInfo) ([]*api.TaskInfo, int) {
+		var victims []*api.TaskInfo
+		allocations := map[api.QueueID]*api.Resource{}
+
+		for _, reclaimee := range reclaimees {
+			job := ssn.Jobs[reclaimee.Job]
+			attr := cp.queueOpts[job.Queue]
+
+			if _, found := allocations[job.Queue]; !found {
+				allocations[job.Queue] = attr.allocated.Clone()
+			}
+			allocated := allocations[job.Queue]
+			if allocated.LessPartly(reclaimer.Resreq, api.Zero) {
+				klog.V(3).Infof("Failed to allocate resource for Task <%s/%s> in Queue <%s>, not enough resource.",
+					reclaimee.Namespace, reclaimee.Name, job.Queue)
+				continue
+			}
+
+			exceptReclaimee := allocated.Clone().Sub(reclaimee.Resreq)
+			if !allocated.LessEqual(attr.deserved, api.Zero) && attr.guarantee.LessEqual(exceptReclaimee, api.Zero) {
+				allocated.Sub(reclaimee.Resreq)
+				victims = append(victims, reclaimee)
+			}
+		}
+		klog.V(4).InfoS("Victims from capacity plugin", "victims", victims, "reclaimer", reclaimer)
+		return victims, util.Permit
+	})
+
+	ssn.AddPreemptiveFn(cp.Name(), func(obj interface{}) bool {
+		queue := obj.(*api.QueueInfo)
+		attr := cp.queueOpts[queue.UID]
+
+		canReclaim := !attr.deserved.LessEqual(attr.allocated, api.Zero)
```

Comment:
Mybe all dimentions resource is large than deserved can be regarded as not reclaimable.

Reply:
Yeah，you are right.